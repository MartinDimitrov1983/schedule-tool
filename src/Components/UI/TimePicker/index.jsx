import React from 'react';
import PropTypes from 'prop-types';
import { Close } from '../../../Assets';
import { TimePicker as DefaultTimePicker } from 'react-time-picker';
import 'react-time-picker/dist/TimePicker.css';
import 'react-clock/dist/Clock.css';
import { TIME } from '../../../Constants';
import styles from './index.module.css';

const TimePicker = ({
  isTemplate,
  isCompleted,
  onChange,
  value,
  onRemoveClick,
}) => {
  const className = isCompleted
    ? styles.completed
    : isTemplate
      ? styles.template
      : styles.autoComplete;
  return (
    <div className={styles.container}>
      <DefaultTimePicker
        className={`${className} ${styles.wrapper}`}
        onChange={onChange}
        value={value}
        format={TIME}
        clearIcon={null}
        disableClock
      />
      {value && isCompleted && (
        <Close
          className={`${styles.icon} ${className}`}
          onClick={onRemoveClick}
        />
      )}
    </div>
  );
};

TimePicker.propTypes = {
  isTemplate: PropTypes.bool,
  isCompleted: PropTypes.bool,
  onChange: PropTypes.func,
  value: PropTypes.string,
  onRemoveClick: PropTypes.func,
};

export default TimePicker;
