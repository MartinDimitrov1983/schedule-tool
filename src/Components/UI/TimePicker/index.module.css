.container {
  position: relative;
  width: 100%;
}
.wrapper {
  border: 1px solid;
  width: 100%;
  /* min-width: 150px; */
  min-height: 40px;
  background-color: white;
}

.wrapper div {
  display: flex;
  align-items: center;
  justify-content: center;
}

.template {
  border-color: #e11bb6;
  fill: #e11bb6;
}

.autoComplete {
  border-color: #acacac;
  fill: #acacac;
}
.completed {
  fill: black;
}

.template > div {
  border: none;
}

.template input {
  color: #e11bb6;
}

.autoComplete input {
  color: #acacac;
}

.template input:invalid {
  background: white !important;
}

.template input::placeholder {
  color: #e11bb6;
}
.template span {
  color: #e11bb6;
}

.autoComplete span {
  color: #acacac;
}

.wrapper button {
  position: absolute;
  right: -4px;
}
.wrapper div {
  display: flex;
  align-items: center;
  justify-content: center;
}

.icon {
  height: 20px;
  width: 20px;
  /* fill: #acacac; */
  position: absolute;
  right: 7px;
  top: 11px;
}
