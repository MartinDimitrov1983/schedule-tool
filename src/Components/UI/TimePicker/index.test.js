import React from 'react';
import { render } from '@testing-library/react';

import TimePicker from './index';

test('Snapshot TimePicker', () => {
  const { asFragment } = render(
    <TimePicker
      isTemplate={true}
      isCompleted={true}
      onChange={() => {}}
      value="12:00"
      onRemoveClick={() => {}}
    />,
  );
  expect(asFragment()).toMatchSnapshot();
});
