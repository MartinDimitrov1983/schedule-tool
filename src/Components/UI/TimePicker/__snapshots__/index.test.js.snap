// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot TimePicker 1`] = `
<DocumentFragment>
  <div
    class="container"
  >
    <div
      class="react-time-picker react-time-picker--closed react-time-picker--enabled completed wrapper"
    >
      <div
        class="react-time-picker__wrapper"
      >
        <div
          class="react-time-picker__inputGroup"
        >
          <input
            hidden=""
            name="time"
            step="60"
            style="visibility: hidden; position: absolute; z-index: -999;"
            type="time"
            value="12:00"
          />
          <input
            autocomplete="off"
            class="react-time-picker__inputGroup__input react-time-picker__inputGroup__hour"
            data-input="true"
            inputmode="numeric"
            max="23"
            min="0"
            name="hour24"
            placeholder="--"
            type="number"
            value="12"
          />
          <span
            class="react-time-picker__inputGroup__divider"
          >
            :
          </span>
          <span
            class="react-time-picker__inputGroup__leadingZero"
          >
            0
          </span>
          <input
            autocomplete="off"
            class="react-time-picker__inputGroup__input react-time-picker__inputGroup__minute react-time-picker__inputGroup__input--hasLeadingZero"
            data-input="true"
            inputmode="numeric"
            max="59"
            min="0"
            name="minute"
            placeholder="--"
            type="number"
            value="0"
          />
        </div>
      </div>
    </div>
    <svg
      class="icon completed"
    >
      close.svg
    </svg>
  </div>
</DocumentFragment>
`;
